src/api/http.js

```diff
@@ -0,0 +1,34 @@
+import merge from 'deepmerge'
+
+export default {
+  defaultOptions: {
+    headers: {
+      'content-type': 'application/json'
+    }
+  },
+  listeners: {
+    request: [],
+    response: []
+  },
+  clearInterceptors: function() {
+    this.listeners.request.splice(0, this.listeners.request.length)
+    this.listeners.response.splice(0, this.listeners.response.length)
+  },
+  addRequestInterceptor: function(listener) {
+    this.listeners.request.push(listener)
+  },
+  addResponseInterceptor: function(listener) {
+    this.listeners.response.push(listener)
+  },
+  fetch: async function(url, overrideOptions = {}) {
+    const options = merge(this.defaultOptions, overrideOptions)
```

Comment:
This is real dumb, but I had no idea `functions` in an object used it as their `this`.

Reply:
@tstodter I think they only work this way if invoked when referencing their parent object like `parentObject.method()` or `this.method()`.  Otherwise, I don't think they get implicitly scoped.